src/repr/src/row/encoding2.rs

```diff
@@ -1201,6 +1181,28 @@ impl ColumnDecoder<Row> for RowColumnarDecoder {
         };
         nullability.is_null(idx)
     }
+
+    fn stats(&self) -> ColumnarStats {
+        let stats = OptionStats {
```

Comment:
Stuff like this in retrospect strikes me as a bit weird - what does the `ColumnarStats::nulls` presence or absence mean if we're always filling it out for every row? - but in any case I've tried to match the semantics of the encoder-time stats.

Reply:
Currently `RowColumnarDecoder` always returns `OptionStats` because the "ok" column of `SourceData` is always nullable. Where the presence or absence of `ColumnarStats::nulls` comes into play is for the inner columns. Where the absence indicates the column is non-nullable.

We could change `RowColumnarDecoder` to be optionally nullable and then return just `StructStats` when non-nullable instead of always returning `OptionStats`?

Reply:
> Currently RowColumnarDecoder always returns OptionStats because the "ok" column of SourceData is always nullable.

For sure. I think some of the weirdness is that despite Row having a first-class codec/schema/decoder impl, we only expect to use it from inside SourceData, and some of the choices are specialized to that case.

Anyways, what you say could totally make sense, but I'm content to leave it until things have stabilized a bit. Just wanted to mention the nonobvious thing in my PR before the reviewers find it on their own...